# Model Evaluation

In [2]:
# Model Comparison Table (reprinted here for ease of information)
import pandas as pd

comparison = pd.DataFrame({
    "Model": ["Logistic Regression", "Random Forest"],
    "Accuracy": [0.7446, 0.7373],
    "Precision (weighted)": [0.7415, 0.7347],
    "Recall (weighted)": [0.7446, 0.7373],
    "F1 Score (weighted)": [0.7402, 0.7317]
})

comparison

,Model,Accuracy,Precision (weighted),Recall (weighted),F1 Score (weighted)
0,Logistic Regression,0.7446,0.7415,0.7446,0.7402
1,Random Forest,0.7373,0.7347,0.7373,0.7317


## Logistic Regression Evaluation

1. Overall Performance:

* Accuracy: 74%
* Logistic Regression performs well on this dataset, with an overall accuracy of 74%, meaning it correctly classifies most instances.

2. Best Performing Class:

* Failure Class:
    * Recall: 0.83 (high)
    * Precision: 0.76 (strong)

* Logistic Regression is particularly good at identifying failed startups. It has both high recall and precision for this class, meaning it correctly identifies failures without many false positives.

3. Weakest Performing Class:

* IPO Class:
    * Recall: 0.27 (very low)
    * Precision: 0.65 (decent but less accurate)

* The IPO class is the weakest, with very low recall. This means that the model misses many actual IPO cases due to the class imbalance, as there are fewer IPO startups in the dataset.

4. Conclusion:

* Strength: Logistic Regression is very strong at predicting failed startups and has decent performance overall.
  
* Weakness: The model struggles with detecting IPO startups due to class imbalance. Although it performs better than Random Forest on IPO recall, it still misses most IPOs.

## Random Forest Evaluation

1. Overall Performance:
   
* Accuracy: 73.7%
* Similar to Logistic Regression, Random Forest achieves about 74% accuracy, indicating it performs reasonably well at predicting startup outcomes.

2. Best Performing Class:

* Failure Class:
    * Recall: 0.82 (high)
    * Precision: 0.76 (strong)

* Like Logistic Regression, Random Forest is good at identifying failed startups with high recall and precision.

3. Weakest Performing Class:

* IPO Class:
    * Recall: 0.16 (very low)
    * Precision: 0.69 (decent)

* The IPO class again shows poor performance, with a very low recall. Random Forest misses even more IPO startups than Logistic Regression, which may be due to its difficulty with rare classes and the class imbalance.

4. Conclusion:

* Strength: Random Forest performs similarly to Logistic Regression for common classes like Failure but struggles more with the IPO class.
  
* Weakness: IPO detection is very weak, and the model doesn't outperform Logistic Regression significantly. Class imbalance remains a major challenge for detecting IPOs.

## Feature Importance (Bar Chart) Evaluation

1. Top Features:
* Revenue: Most important factor for IPO.
* Product Traction: High user engagement is key for IPO success.

2. Moderate Features:
* Team Size, Burn Rate, Market Size, Founder Experience, Funding Rounds: Important but less decisive.

3. Least Important Features:
* Sector, Founder Background, Investor Type: Minimal impact on IPO predictions.

## SHAP Beeswarm Plot (IPO Class - rare)

1. Top Contributors:
* Revenue: Strongly pushes startups towards IPO.
* Product Traction Users: Increases IPO likelihood.

2. Moderate Contributors:
* Funding Rounds, Founder Experience Years, Team Size: Slight positive impact on IPO prediction.

3. Least Impactful:
* Sector, Burn Rate, Market Size, Investor Type, Founder Background: Little effect on IPO classification.

| **Aspect**                   | **Feature Importance**                                                | **SHAP (IPO class - rare)**                                                    |
| ---------------------------- | --------------------------------------------------------------------- | ------------------------------------------------------------------------------ |
| **Top Contributors for IPO** | Revenue, Product Traction                                             | Revenue, Product Traction                                                      |
| **Moderate Contributors**    | Team Size, Burn Rate, Market Size, Founder Experience, Funding Rounds | Funding Rounds, Founder Experience Years, Team Size                            |
| **Least Impactful**          | Sector, Founder Background, Investor Type                             | Sector, Burn Rate, Market Size, Investor Type, Founder Background              |
| **Strengths**                | Big-picture insights on feature importance                            | Individual-level explanations for IPO predictions                              |
| **Use Case**                 | Shows overall feature importance for IPO                              | Explains why a startup is classified as IPO at the individual prediction level |


## Final Takeaways

* **Model Performance**

  * Logistic Regression (**74.46%**) slightly outperforms Random Forest (**73.73%**)
  * Both models show similar overall performance → dataset has **moderate predictability**

* **No Clear Best Model**

  * Logistic Regression performs slightly better overall
  * Random Forest does not provide a meaningful improvement

* **Class Imbalance Impact (Key Issue)**

  * Both models perform well on **Failure** and **Acquisition**
  * Both struggle heavily with **IPO prediction**

    * Logistic Regression IPO recall: **0.27**
    * Random Forest IPO recall: **0.16**

* **Best vs Worst Class**

  * **Best:** Failure (recall ≈ 0.82–0.83)
  * **Worst:** IPO (consistently low recall across both models)

* **Key Feature Insights**

  * **Revenue** = strongest predictor of startup success/IPO
  * **Product Traction** = second most important factor
  * Other features (team size, burn rate, market size) = moderate impact
  * Founder/investor features = weak predictive power

* **SHAP vs Feature Importance**

  * SHAP confirms Feature Importance results
  * Revenue + Product Traction dominate both global and IPO-specific predictions
  * SHAP adds **instance-level explanation** (why a specific startup is predicted as IPO)

* **Overall Conclusion**

  * Models are effective for general classification but **weak for IPO prediction**
  * IPO remains difficult due to **severe class imbalance**
  * Success is mainly driven by **financial and user traction metrics** rather than founder/investor attributes
